packages/core/src/domain/context/contextUtils.ts

```diff
@@ -0,0 +1,32 @@
+import type { Context } from '../../tools/serialisation/context'
+import { display } from '../../tools/display'
+import { getType } from '../../tools/utils/typeUtils'
+import { assign } from '../../tools/utils/polyfills'
+
+/**
+ * Clone input data and ensure known context properties
+ * are strings, as defined here:
+ * https://docs.datadoghq.com/logs/log_configuration/attributes_naming_convention/#user-related-attributes
+ */
+export function sanitizeContext(newContext: Context, keys: string[]): Context {
+  // We shallow clone only to prevent mutation of context data.
+  const context = assign({}, newContext)
+  keys.forEach((key) => {
+    if (key in context) {
+      /* eslint-disable @typescript-eslint/no-base-to-string */
+      context[key] = String(context[key])
+    }
+  })
+  return context
+}
+
+/**
+ * Simple check to ensure an object is a valid context
+ */
+export function checkContext(newContext: Context): boolean {
```

Comment:
Would it make sense for the signature of this function to be this instead?
```suggestion
export function checkContext(maybeContext: unknown): maybeContext is Context {
```

That way you wouldn't need the `as Context` type assertions you're currently applying at call sites; instead, you could use this function to narrow the type of a value to `Context` when it can be treated as one.